featuretools/feature_base/feature_descriptions.py

```diff
@@ -0,0 +1,153 @@
+        slice_num = feature.n
+    if feature.primitive in primitive_templates or feature.primitive.name in primitive_templates:
+        template_override = (primitive_templates.get(feature.primitive) or
+                             primitive_templates.get(feature.primitive.name))
+    primitive_description = feature.primitive.get_description(input_descriptions,
+                                                              slice_num=slice_num,
+                                                              template_override=template_override)
+    if isinstance(feature, ft.feature_base.FeatureOutputSlice):
+        feature = feature.base_feature
+
+    # Generate groupby phrase if applicable
+    groupby = ''
+    if isinstance(feature, ft.AggregationFeature):
+        groupby_description = get_aggregation_groupby(feature, feature_descriptions)
+    if groupby_description is not None:
+        if groupby_description.startswith('the '):
+            groupby_description = groupby_description[4:]
+        groupby = "for each {}".format(groupby_description)
+
+    # Generate aggregation entity phrase with use_previous
+    entity_description = ''
+    if isinstance(feature, ft.AggregationFeature):
+        if feature.use_previous:
+            entity_description = "of the previous {} of ".format(
+                feature.use_previous.get_name().lower())
+        else:
+            entity_description = "of all instances of "
+        entity_description += '"{}"'.format(feature.relationship_path[-1][1].child_entity.id)
+
+    # Generate where phrase
+    where = ''
+    if hasattr(feature, 'where') and feature.where:
+        where_col = generate_description(feature.where.base_features[0],
+                                         feature_descriptions,
+                                         primitive_templates)
+        where = 'where {} is {}'.format(where_col, feature.where.primitive.value)
+
+    # Join all parts of template
+    description_template = [primitive_description, entity_description, where, groupby]
+    description = " ".join([phrase for phrase in description_template if phrase != ''])
+
+    return description
+
+
+def get_direct_description(feature):
+    direct_description = ' the instance of "{}" associated with this ' \
+        'instance of "{}"'.format(feature.relationship_path[-1][1].parent_entity.id,
+                                  feature.entity_id)
+    base_features = feature.base_features
+    # shortens stacked direct features to make it easier to understand
+    while isinstance(base_features[0], ft.DirectFeature):
+        base_feat = base_features[0]
+        base_feat_description = ' the instance of "{}" associated ' \
+            'with'.format(base_feat.relationship_path[-1][1].parent_entity.id)
+        direct_description = base_feat_description + direct_description
+        base_features = base_feat.base_features
+    direct_description = ' of' + direct_description
```

Comment:
Also wondering if this should be `for` instead of `of`. If we changed this along with the change above the output of  the first test in `test_direct_description` would change from:
```
The "loves_ice_cream" of the instance of "customers" associated with this instance of "sessions".
```

to:
```
"loves_ice_cream" for the instance of "customers" associated with this instance of "sessions".
```

Again, not say we should change, just an idea.